src/Restyler/Config.hs

```diff
@@ -207,21 +207,29 @@ readConfigSource = \case
         if exists then Just <$> readFileBS path else pure Nothing
     ConfigContent content -> pure $ Just content
 
+readFirstConfigSource
+    :: HasSystem env => [ConfigSource] -> RIO env (Maybe ByteString)
+readFirstConfigSource = findJust . fmap readConfigSource
+
+findJust :: Monad m => [m (Maybe a)] -> m (Maybe a)
+findJust = foldM (\acc x -> maybe x (pure . Just) acc) Nothing
```

Comment:
This is nice, but it's a bit dense to read. I didn't realize this until just now, but I think using the `Alternative` instance for `MaybeT` could be clearer,

```hs
readFirstConfigSource
    :: HasSystem env => [ConfigSource] -> RIO env (Maybe ByteString)
readFirstConfigSource = runMaybeT . asum . map (MaybeT . readConfigSource)
```

IIUC, `asum` is a fold of `<|>` and the semantics of `(<|>)` for `MaybeT` prevent actions on the right hand side from evaluating if you get `Just` our of the left hand side, which is what's important here.

WDYT?

Reply:
I hadn't thought about `Alternative` but it seems appropriate here. I'll try the `asum` suggestion as well.